invenio_rdm_records/services/search_params.py

```diff
@@ -24,3 +24,33 @@ def apply(self, identity, search, params):
         if value is not None and value in [x.value for x in RecordDeletionStatusEnum]:
             search = search.filter("term", **{"deletion_status": value})
         return search
+
+
+class AdminParam(ParamInterpreter):
+    """Evaluates the 'status' parameter."""
+
+    def apply(self, identity, search, params):
+        """Evaluate the status parameter on the search."""
+        value = params.pop("is_admin", None)
+        # Filter prevents from displaying deleted records on mainsite search
+        # deleted records should appear only in admins panel
+        if value is None:
+            search = search.filter(
+                "term", **{"deletion_status": RecordDeletionStatusEnum.PUBLISHED.value}
+            )
+        return search
+
+
+class AdminDraftsParam(ParamInterpreter):
+    """Evaluates the 'is_admin' parameter."""
+
+    def apply(self, identity, search, params):
+        """Evaluate the is_admin parameter on the search."""
+        value = params.pop("is_admin", None)
+        # Filter prevents from other users' drafts from displaying on Moderator's
+        # dashboard
+        if value is None:
+            search = search.filter(
+                "term", **{"parent.access.owned_by.user": current_user.id}
+            )
+        return search
```

Comment:
Comment: I personally find it a bit confusing to have a parameter `is_admin`, it feels like it will give superpowers if it's set to `true`. A part from that, I would personally go for more explicit parameters, dividing this 2 Interpreters in 2 different params maybe something like `deleted` and `all`.

Reply:
What about renaming it to `with_deleted` to indicate that I want the deleted records or not and then return if you have the right permissions?

Reply:
it will have a naming clash with with_deleted argument for details views, I would like to avoid that